socketshark/subscription.py

```diff
@@ -130,8 +180,50 @@ def should_deliver_message(self, data):
                                          reason='order')
             return False
 
+        if not self._should_deliver_message_throttle(data):
+            self.session.trace_log.debug('message filtered', data=data,
+                                         reason='throttle')
+            return False
+
         return True
 
+    async def _schedule_throttled_message(self, when, throttle_key):
+        delay = when - time.time()
+        try:
+            await asyncio.sleep(delay)
+            await self._send_throttled_message(throttle_key)
+        except asyncio.CancelledError:  # Cancelled by unsubscribe
+            self.session.trace_log.debug('throttled message canceled',
+                                         throttle_key=throttle_key)
+        except Exception:
+            self.session.log.exception('unhandled exception when sending '
+                                       'throttled message')
+
+    async def _send_throttled_message(self, throttle_key):
+        # We've unsubscribed meanwhile.
+        if self.name not in self.session.subscriptions:
+            self.session.trace_log.debug('throttled message subscription '
+                                         'invalid', throttle_key=throttle_key)
+            return
+        last_throttle = self.throttle_state.get(throttle_key)
+        if not last_throttle:
```

Comment:
This could use a comment to describe what it is catching

Reply:
I'll probably just remove this and change it to a `self.throttle_state[throttle_key]` lookup since this condition shouldn't happen.

Reply:
I added a few more log statements though I think this particular one should be fine since we're already getting a `message filtered` log.